Xamarin.Forms.Platform.Android/ImageCache.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	/// <summary>
+	/// I setup the access to all the cache elements to be async because
+	/// if I didn't then it was locking up the GC and freezing the entire app
+	/// </summary>
+	class ImageCache
+	{
+		readonly FormsLruCache _lruCache;
+		readonly ConcurrentDictionary<string, SemaphoreSlim> _waiting;
+
+		public ImageCache() : base()
+		{
+			_waiting = new ConcurrentDictionary<string, SemaphoreSlim>();
+			_lruCache = new FormsLruCache();			
+		}
+
+		void Put(string key, TimeSpan cacheValidity, global::Android.Graphics.Bitmap cacheObject)
+		{
+			_lruCache.Put(key, new CacheEntry() { TimeToLive = DateTimeOffset.Now.Add(cacheValidity), Data = cacheObject });
```

Comment:
Should this use `UtcNow`?

Reply:
What does using UtcNow vs Now give us?

Reply:
`Now` is going to calculate your local time -- some timezone math is involved. Also the user's timezone can change?

Here is a good explanation: https://blogs.msdn.microsoft.com/kirillosenkov/2012/01/10/datetime-utcnow-is-generally-preferable-to-datetime-now/

I would pretty much not ever use `Now` unless you are presenting a date/time to a user, and they need to see it in their local time.

Reply:
That's fair. I didn't know Now was calling UtcNow and converting; TIL.

Reply:
Changed to UtcNow